Mathlib/Computability/AkraBazzi/GrowsPolynomially.lean

```diff
@@ -0,0 +1,340 @@
+/-
+Copyright (c) 2023 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import Mathlib.Analysis.Asymptotics.AsymptoticEquivalent
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+
+/-!
+# Akra-Bazzi theorem: The polynomial growth condition
+
+This file defines and develops an API for the polynomial growth condition that appears in the
+statement of the Akra-Bazzi theorem: for the Akra-Bazzi theorem to hold, the function `g` must
+satisfy the condition that c₁ g(n) ≤ g(u) ≤ c₂ g(n), for u between b*n and n for any constant
+b ∈ (0,1).
+-/
+
+set_option autoImplicit false
+
+open Finset Real Filter Asymptotics BigOperators
+
+namespace AkraBazziRecurrence
+
+/-- The growth condition that the function `g` must satisfy for the Akra-Bazzi theorem to apply.
```

Comment:
Yes, you're correct, and I actually considered defining it for `b = 1/2`. I went with this version because the `b = 1/2` version doesn't make any of the proofs easier, and it's the version we actually want to use anyway. I could switch to that definition and prove the equivalence if you think it's worth it.